addons/udes_stock/models/stock_picking.py

```diff
@@ -627,25 +635,37 @@ def _create_batch(self, pickings):
         PickingBatch = self.env["stock.picking.batch"]
         PickingBatch.create({"picking_ids": [(6, 0, pickings.ids)]})
 
-    def _prepare_move(self, pickings, products_info, **kwargs):
+    def _prepare_move(self, pickings, products_info, extra_product_info_keys=None, **kwargs):
         """Return a list of the move details to be used later in creation of the move(s).
         The purpose of this is to allow for multiple moves to be created at once.
 
         :args:
             - pickings: iterable of picking objects to be assigned to the moves
             - products_info: list(list(dict)) with dict of product, uom_qty and uom_id
             If uom_id is not specified, the product UoM is  used.
+            - extra_product_info_keys: list(str) or None: Additional keys to unpack to field values
+            which should be looked up in products_info. This allows us to pass extra fields to move
+            creation when calling create_picking().
 
         :returns:
             Move_values: list(dict)
         """
         move_values = []
+        if extra_product_info_keys is None:
+            extra_product_info_keys = []
         for i, picking in enumerate(pickings):
             for product_info in products_info[i]:
                 product = product_info.get("product")
                 uom_qty = product_info.get("uom_qty") or product_info.get("qty")
                 uom_id = product_info.get("uom_id") or product.uom_id.id
                 expected_date = product_info.get("expected_date")
+                # Unpack additional keys from product_info for when we need to
+                # set additional fields on the move when calling create_picking()
+                kwargs_by_proxy = {
+                    k: v
+                    for k, v in product_info.items()
+                    if k in extra_product_info_keys
+                }
                 vals = {
```

Comment:
Maybe better create vals in a method which can be easily overriden smth like:

vals = picking.get_move_vals_from_product_info(product_info)

Reply:
Yeah this can probably work. Didn't think of abstracting the inner part, i'll make the change